sqlmesh/core/plan/definition.py

```diff
@@ -292,22 +292,15 @@ def set_choice(self, snapshot: Snapshot, choice: SnapshotChangeCategory) -> None
                 f"A choice can't be changed for the existing version of model '{snapshot.name}'."
             )
 
-        snapshot.change_category = choice
-        if choice in (
-            SnapshotChangeCategory.BREAKING,
-            SnapshotChangeCategory.NON_BREAKING,
-        ):
-            snapshot.set_version()
-        else:
-            snapshot.set_version(snapshot.previous_version)
+        snapshot.categorize_as(choice)
 
         for child in self.indirectly_modified[snapshot.name]:
             child_snapshot = self.context_diff.snapshots[child]
 
             if choice == SnapshotChangeCategory.BREAKING:
-                child_snapshot.set_version()
+                child_snapshot.categorize_as(SnapshotChangeCategory.INDIRECT_BREAKING)
             else:
-                child_snapshot.set_version(child_snapshot.previous_version)
+                child_snapshot.categorize_as(SnapshotChangeCategory.INDIRECT_FORWARD_ONLY)
```

Comment:
i noticed indirect-non-breaking is not used anywhere

Reply:
See comment: https://github.com/TobikoData/sqlmesh/pull/763#discussion_r1177086530